File: solutions/javascript/bird-watcher/1/bird-watcher.js
// @ts-check

export function totalBirdCount(birdsPerDay) {
  var count = 0;
  for(let i=0; i<birdsPerDay.length; i++){
    count += birdsPerDay[i]
  }
  return count;
}

export function birdsInWeek(birdsPerDay, week) {
  const start = (week - 1) * 7;
  let sum = 0;

  // Sum up to 7 days, but stop if we run out of array
  for (let i = 0; i < 7; i++) {
    const idx = start + i;
    if (idx >= birdsPerDay.length) break;
    sum += birdsPerDay[idx];
  }

  return sum;
}

// export function fixBirdCountLog(birdsPerDay) {
//   var fixedBirdCountPerDay = [];
//   for (let i = 0; i < birdsPerDay.length; i++) {
//     if (i % 2 == 0) fixedBirdCountPerDay.push(birdsPerDay[i] + 1);
//     else fixedBirdCountPerDay.push(birdsPerDay[i]);
//   }
//   return fixedBirdCountPerDay;
// }


export function fixBirdCountLog(birdsPerDay) {
  for (let i = 0; i < birdsPerDay.length; i += 2) {
    birdsPerDay[i]++;
  }
  return birdsPerDay;
}

File: solutions/go/bird-watcher/1/bird_watcher.go
package birdwatcher

func TotalBirdCount(birdsPerDay []int) int {
	count := 0
	for i := 0; i < len(birdsPerDay); i++ {
		count += birdsPerDay[i]
	}
	return count
}

func BirdsInWeek(birdsPerDay []int, week int) int {

	start := (week - 1) * 7
	end := start + 7
	count := 0

	for i := start; i < end; i++ {
		count += birdsPerDay[i]
	}

	return count
}

func FixBirdCountLog(birdsPerDay []int) []int {
	for i := 0; i < len(birdsPerDay); i += 2 {
		birdsPerDay[i]++
	}
	return birdsPerDay
}
